code/game/objects/structures/foodcart.dm

```diff
@@ -139,6 +175,51 @@
 			user.put_in_hands(food_slots[6])
 			to_chat(user, "<span class='notice'>You take [food_slots[6]] from [src].</span>")
 			food_slots[6] = null
+	if(href_list["f7"])
```

Comment:
Absolutely fucking not. Refactor this to make it use one single eject href, or TGUI it. Repetition like this is not happening.